### PR TITLE
KAA-943: Selective travis run

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @rasendubi @Dubland

### CODE REVIEW

File: client/client-multi/client-c/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
```
Comment (rasendubi) [inline client/client-multi/client-c/deploy.sh]: Use `/bin/sh`

File: client/client-multi/client-c/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+case $1 in
+    install)
+        curl https://nixos.org/nix/install | sh
+        source $HOME/.nix-profile/etc/profile.d/nix.sh
+        sudo mkdir /etc/nix
+        sudo sh -c 'echo "build-max-jobs = 4" > /etc/nix/nix.conf'
+        nix-shell -Q --arg cc3200Support false --pure --run true
```
Comment (rasendubi) [inline client/client-multi/client-c/deploy.sh]: Use `${@:2}` instead of `--arg cc3200Support`

**Comment by rasendubi** [comment]
I'd better split the script to two separate scripts (setup / build). The name `deploy.sh` is especially funny because it doesn't deploy anything

File: client/client-multi/client-c/deploy.sh (hunk 1 above)
Comment (RostakaGmfun) [inline client/client-multi/client-c/deploy.sh]: travis runs scripts in bash.
Furthermore, argument array slicing is a `bash` feature, not supported by `sh`.

File: .travis.yml [hunk 3]
```diff
@@ -3,17 +3,15 @@ python: "3.4"
...
-  - bash <(curl -sS https://nixos.org/nix/install)
-  - source $HOME/.nix-profile/etc/profile.d/nix.sh
-
-  - sudo mkdir /etc/nix
-  - sudo sh -c 'echo "build-max-jobs = 4" > /etc/nix/nix.conf'
-
-  - cd client/client-multi/client-c
-  - nix-shell -Q --arg cc3200Support false --pure --run true
-  - cd ../../..
+    | if test -n "$(git diff --name-only ${TRAVIS_COMMIT_RANGE/.../..} -- client/client-multi/client-c)"; then
```
Comment (rasendubi) [inline .travis.yml]: Move `|` to previous line, use proper indentation.

File: client/client-multi/client-c/scripts/build.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#! /bin/bash
+
+nix-shell ${@:2} --pure --run "make"
+nix-shell ${@:2} --pure --run "./build.sh test"
```
Comment (rasendubi) [inline client/client-multi/client-c/scripts/build.sh]: Now you should use `$@`

File: .travis.yml [hunk 5]
```diff
@@ -2,18 +2,16 @@ language: python
 python: "3.4"
 sudo: required
 dist: trusty
-install:
-  - bash <(curl -sS https://nixos.org/nix/install)
-  - source $HOME/.nix-profile/etc/profile.d/nix.sh
+install: |
+    if test -n "$(git diff --name-only ${TRAVIS_COMMIT_RANGE/.../..} -- client/client-multi/client-c)"; then
+    cd client/client-multi/client-c
```
Comment (rasendubi) [inline .travis.yml]: Indent

File: client/client-multi/client-c/scripts/install.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#! /bin/bash
+
+curl https://nixos.org/nix/install | sh
+source $HOME/.nix-profile/etc/profile.d/nix.sh
+sudo mkdir /etc/nix
+sudo sh -c 'echo "build-max-jobs = 4" > /etc/nix/nix.conf'
+nix-shell -Q ${@:2} --pure --run true
```
Comment (rasendubi) [inline client/client-multi/client-c/scripts/install.sh]: Same here

File: client/client-multi/client-c/scripts/install.sh [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#! /bin/bash
```
Comment (rasendubi) [inline client/client-multi/client-c/scripts/install.sh]: `#!/bin/sh`

File: client/client-multi/client-c/scripts/build.sh [hunk 8]
```diff
@@ -0,0 +1,4 @@
+#! /bin/bash
```
Comment (rasendubi) [inline client/client-multi/client-c/scripts/build.sh]: `#!/bin/sh`

File: .travis.yml [hunk 9]
```diff
@@ -2,18 +2,16 @@ language: python
...
 sudo: required
 dist: trusty
-install:
-  - bash <(curl -sS https://nixos.org/nix/install)
-  - source $HOME/.nix-profile/etc/profile.d/nix.sh
+install: |
+  if test -n "$(git diff --name-only ${TRAVIS_COMMIT_RANGE/.../..} -- client/client-multi/client-c)"; then
+    cd client/client-multi/client-c &&
+    ./scripts/install.sh --arg cc3200Support false &&
+    . $HOME/.nix-profile/etc/profile.d/nix.sh
```
Comment (rasendubi) [inline .travis.yml]: Missed &&

**Comment by rasendubi** [comment]
LGTM

**Pull request merged**